kie-server-parent/kie-server-router/kie-server-router-proxy/src/test/java/org/kie/server/router/repository/FileRepositoryTest.java

```diff
@@ -213,4 +213,15 @@ public void onConfigurationReloaded() {
         repoWithWatcher.close();
         repoWithWatcher.clean();
     }
+
+    @Test
+    public void testFileCreation() throws Exception {
+
+        File repositoryDirectory = new File("target" + File.separator + UUID.randomUUID().toString());
+        Configuration configuration = new Configuration();
+        ConfigurationMarshaller marshaller = new ConfigurationMarshaller();
+
+        ConfigFileWatcher configFileWatcher = new ConfigFileWatcher(repositoryDirectory.getParentFile().getAbsolutePath(), marshaller, configuration);
+        assertNotNull(configFileWatcher);
```

Comment:
not sure if this test really test if the file exists, it on tests the object, and at this point it will be never null even if the cfg file is not correctly wriitten.

probably reflection would be a way to test it, @elguardian wdyt?

Reply:
assertNotNull(configFileWatcher); <- this is always true.
1) He will need to check first that the directory contains the file being created.
assertFileExists(configurationFile)

2) unmarshall the file and check the defaults are equals to configuration
assertEquals(configuration, unmarshalledConfigurationFromFile)

Those would be good condition guards.